File: 485 Max Consecutive Ones.c

int findMaxConsecutiveOnes(int* nums, int numsSize){
    bool bIn = false;
    int maxLen = 0;
    int i;
    int from;
    
    for (i = 0; i < numsSize; i++) {
        if (nums[i]) {
            if (!bIn) {
                from = i;
                bIn = true;
            }
        }
        else {
            if (bIn) {
                int len = i - from;
                if (len > maxLen) {
                    maxLen = len;
                }
            }
            bIn = false;
        }
    }

    if (bIn) {
        int len = i - from;
        if (len > maxLen) {
            maxLen = len;
        }
    }
    
    return maxLen;
}

File: 485 Max Consecutive Ones.java
class Solution {
    public int findMaxConsecutiveOnes(int[] nums) {
        int result = 0;
        int temp = 0;
        
        for (int i = 0 ; i < nums.length ; i++) {
            
            if(nums[i] == 0) {
                result = Math.max(result, temp);
                temp = 0;
                continue;
            }
            
            temp++;            
        }
        
        result = Math.max(result, temp);
        
        return result;
    }
}
